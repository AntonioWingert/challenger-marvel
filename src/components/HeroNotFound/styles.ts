import { styled } from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 100vw;
  height: 100vh;
  background-color: #ccc;

  a {
    display: flex;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    color: ${({theme}) => theme.colors["secondary"]};
    background-color: ${({theme}) => theme.colors["bg-primary"]};
    width: 30%;
    height: 4rem;
    border-radius: 0.5rem;
    font-size: 1.5rem;
  }

  img {
    width: 20%;
    height: auto;
  }

  @media (max-width: 768px) {
    a {
      width: 60%;
      font-size: 1rem;
    }

    img {
      width: 80%;
      height: auto;
    }
  }
`;