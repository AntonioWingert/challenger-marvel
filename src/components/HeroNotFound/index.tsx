import CustomTitle from "../CustomTitle";
import * as S from "./styles";

import SpiderImage from "/spider-man.png";


import { Link } from "react-router-dom";

export default function HeroNotFound() {
  return (
    <>
      <S.Container>
        <img src={SpiderImage} />
        <CustomTitle title="Herói não encontrado" size="lg" />
        <Link to='/'>Voltar</Link>
      </S.Container>
    </>
  );
}