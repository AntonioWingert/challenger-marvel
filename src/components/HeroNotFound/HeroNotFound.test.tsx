import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import HeroNotFound from '.';

describe('<HeroNotFound />', () => {
  it('should render correctly with default props', () => {
    const { container } = render(
      <HeroNotFound />
    );

    expect(screen.getByText(/Herói não encontrado/i)).toBeVisible();
    expect(screen.getByAltText('Hero not found')).toBeVisible();
    expect(screen.getByAltText('Hero not found')).toHaveAttribute('src', '/spider-man.png');
    expect(container.firstChild).toMatchSnapshot();
  });
});