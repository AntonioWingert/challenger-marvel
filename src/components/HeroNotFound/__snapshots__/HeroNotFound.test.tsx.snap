// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeroNotFound /> > should render correctly with default props 1`] = `
.c1 {
  font-size: 2.4rem;
  color: #0D0D0D;
  font-weight: 400;
  padding: 32px;
}

.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 100vw;
  height: 100vh;
  background-color: #ccc;
}

.c0 a {
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  color: #F5F5F5;
  background-color: #4A1414;
  width: 30%;
  height: 4rem;
  border-radius: 0.5rem;
  font-size: 1.5rem;
}

.c0 img {
  width: 20%;
  height: auto;
}

@media (max-width: 320px) {
  .c1 {
    font-size: 1.6rem;
  }
}

@media (max-width: 768px) {
  .c0 a {
    width: 60%;
    font-size: 1rem;
  }

  .c0 img {
    width: 80%;
    height: auto;
  }
}

<div
  class="c0"
>
  <img
    alt="Hero not found"
    src="/spider-man.png"
  />
  <h1
    class="c1"
  >
    Herói não encontrado
  </h1>
  <a
    href="/"
  >
    Voltar
  </a>
</div>
`;
